modules/SofaSparseSolver/src/SofaSparseSolver/SparseCholeskySolver.cpp

```diff
@@ -108,10 +109,67 @@ void SparseCholeskySolver<TMatrix,TVector>::invert(Matrix& M)
     A.nz = -1;							// # of entries in triplet matrix, -1 for compressed-col
     cs_dropzeros( &A );
     tmp.resize(A.n);
-    S = cs_schol (&A, order) ;		/* ordering and symbolic analysis */
+    
+    perm.resize(A.n);
+    iperm.resize(A.n);
+
+    fill_reducing_perm(A , perm.data(), iperm.data() ); // compute the fill reducing permutation
+    S = symbolic_Chol( &A ); // symbolic analysis
+    permuted_A = cs_permute( &(A), iperm.data() , perm.data() , 1);
+
+    sofa::helper::ScopedAdvancedTimer factorizationTimer("factorization");
     N = cs_chol (&A, S) ;		/* numeric Cholesky factorization */
 }
 
+template<class TMatrix, class TVector>
+void SparseCholeskySolver<TMatrix,TVector>::fill_reducing_perm(cs A,int * perm,int * invperm)
+{
+    int n = A.n;
+    if(d_applyPermutation.getValue() )
+    {   
+        sofa::type::vector<int> adj,xadj;
+
+        CSR_to_adj( A.n, A.p , A.i , adj, xadj);
+        
+        METIS_NodeND(&n, xadj.data(), adj.data(), NULL, NULL, perm,invperm);
```

Comment:
```suggestion
        METIS_NodeND(&n, xadj.data(), adj.data(), nullptr, nullptr, perm,invperm);
```
Prefer `nullptr` over `NULL`